### PR TITLE
support DNS chaos

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Signed-off-by: xiang <xiang13225080@163.com>

fix https://github.com/chaos-mesh/chaosd/issues/41, this pr just support update the dns server in /etc/resolv.conf

TODO: map specified host to specified IP

```bash
./bin/chaosd attack network dns --help
attack DNS server

Usage:
  chaosd attack network dns [flags]

Flags:
      --dns-server string   update the DNS server in /etc/resolv.conf with this value (default "123.123.123.123")
  -h, --help                help for dns

Global Flags:
      --log-level string   the log level of chaosd, the value can be 'debug', 'info', 'warn' and 'error'
```

### CODE REVIEW

File: pkg/core/network.go [hunk 1]
```diff
@@ -141,6 +151,12 @@ func (n *NetworkCommand) setDefaultForNetworkLoss() {
 	}
 }
 
+func (n *NetworkCommand) setDefaultForNetworkDNS() {
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply pkg/core/network.go:156]: @muse-dev ignore

File: pkg/server/chaosd/network.go [hunk 2]
```diff
@@ -36,24 +36,31 @@ func (networkAttack) Attack(options core.AttackConfig, env Environment) (err err
 		ipsetName string
 	)
 
-	if attack.NeedApplyIPSet() {
-		ipsetName, err = env.Chaos.applyIPSet(attack, env.AttackUid)
-		if err != nil {
-			return errors.WithStack(err)
+	switch attack.Action {
+	case core.NetworkDNSAction:
```
Comment (AsterNighT) [inline pkg/server/chaosd/network.go:40]: That seems a bit weird. Why can't we use some `if attack.NeedApplyDNS()`?

Comment (AsterNighT) [reply pkg/server/chaosd/network.go:40]: But certainly `DNSAttack` will have `attack.NeedApplyIpset() == false`, so the original way would work properly?

Comment (WangXiangUSTC) [reply pkg/server/chaosd/network.go:40]: Yes, it can work, but I think use switch can make the logic more clear

Comment (AsterNighT) [reply pkg/server/chaosd/network.go:40]: That's ok. Rest lgtm

**Review by AsterNighT — APPROVED** [review]

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Pull request merged**